packages/fxa-auth-server/lib/routes/index.js

```diff
@@ -200,8 +203,40 @@ module.exports = function (
     newsletters,
     linkedAccounts
   );
+
+  function optionallyIgnoreTrace(fn) {
+    return async function (request) {
+      // Only authenticated routes or routes that specify an uid/email
```

Comment:
If a route doesn't have any user info present, is it even possible to violate the opt-out flag? If the answer is 'no', maybe they could still trace the route? If the answer here is 'yes' or 'maybe', then this sounds good to me.

Reply:
Ah I've updated this variable to better describe this (`canOptOut`). Routes that don't have user info will still be traced.